test/IntegrationTests/Helpers/MockMetricsCollector.cs

```diff
@@ -180,36 +172,29 @@ public void AssertResourceExpectations(TimeSpan? timeout = null)
             throw new InvalidOperationException("Expectations were not set");
         }
 
-        if (_expectations.Count > 0)
-        {
-            throw new InvalidOperationException("Currently you can only assert for metrics or resouce attributes");
-        }
-
         timeout ??= DefaultWaitTimeout;
-        var cts = new CancellationTokenSource();
 
         try
         {
-            cts.CancelAfter(timeout.Value);
-            var resourceMetrics = _metrics.Take(cts.Token); // get the metrics snapshot, each contains the same resources
-            AssertResourceMetrics(_resourceExpectations, resourceMetrics);
+            if (!_resourceAttributesEvent.WaitOne(timeout.Value))
+            {
+                FailResourceMetrics(_resourceExpectations, null);
+                return;
+            }
+
+            AssertResourceMetrics(_resourceExpectations, _resourceAttributes);
         }
         catch (ArgumentOutOfRangeException)
         {
             // CancelAfter called with non-positive value
             FailResourceMetrics(_resourceExpectations, null);
         }
-        catch (OperationCanceledException)
```

Comment:
Just checking, the test framework itself knows to handle OperationCanceledException?

Reply:
Nothing should throw it as I removed the uasge of CancellationTokenSource in this place

Reply:
`var metrics = _metricsSnapshots.Take(cts.Token);` this one?

Reply:
nvm... It's still catching there, my bad 😊

Reply:
That call is in a different method now, and that exception is still handled in that method. The method that grabs the resource attributes uses `_resourceAttributesEvent.WaitOne(timeout.Value)` which does not throw that exception according to the docs.